vendor/k8s.io/kubernetes/pkg/cloudprovider/providers/aws/aws_test.go

```diff
@@ -69,6 +70,25 @@ func (m *MockedFakeEC2) DescribeSecurityGroups(request *ec2.DescribeSecurityGrou
 }
 
 func (m *MockedFakeEC2) CreateVolume(request *ec2.CreateVolumeInput) (*ec2.Volume, error) {
+	// mock requires stable input, and in CreateDisk we invoke buildTags which uses
+	// a map to create tags, which then get converted into an array. This leads to
+	// unstable sorting order which confuses mock. Sorted tags are not needed in
+	// regular code, but are a must in tests here:
+	for i := 0; i < len(request.TagSpecifications); i++ {
```

Comment:
I don't think this is enough. This is just basically sorting the tags in the `request`. Though, the real `CreateVolume` is still constructing the list of tags based on the order of iterating through map keys. We either need to sort the list of tags right before `CreateVolume` is called. Or, just specifiy `awsServices.ec2.(*MockedFakeEC2).On("CreateVolume", request).Return(volume, nil)` on more time just with different order of tags in `request`.

Reply:
Real `CreateVolume` is called with the following code:
```go
	request := &ec2.CreateVolumeInput{}
	...
	tags := volumeOptions.Tags
	tags = c.tagging.buildTags(ResourceLifecycleOwned, tags)

	var tagList []*ec2.Tag
	for k, v := range tags {
		tagList = append(tagList, &ec2.Tag{
			Key: aws.String(k), Value: aws.String(v),
		})
	}
	request.TagSpecifications = append(request.TagSpecifications, &ec2.TagSpecification{
		Tags:         tagList,
		ResourceType: aws.String(ec2.ResourceTypeVolume),
	})

	response, err := c.ec2.CreateVolume(request)
```

Ordering of tags in the list (i.e. `tagList`) is still constructed randomly based on the order in which `tags` (map[string]string) is iterated. Signature of `buildTags` method: `buildTags(lifecycle ResourceLifecycle, additionalTags map[string]string) map[string]string`.

Reply:
Discussed this on slack, this is working as expected, iow. before above `CreateVolume` is being called we are sorting the tags to have stable input which match those that are in `TestCreateDisk`. I've added additional comment in `TestCreateDisk` next to `Tags` to ensure we have those values sorted.